### ISSUE TITLE
Derived value classes with type parameters exhibit odd behavior with classOf

### ISSUE THREAD

**Issue by hrhino** [scala/bug#10551]
They have a bit of an existential crisis:

```
scala> final class Me[X](val mine: X) extends AnyVal
defined class Me

scala> classOf[Me[Int]]
res0: Class[Me[Int]] = class Me

scala> classOf[Me[String]]
res1: Class[Me[String]] = class Me

scala> classOf[Me[_]]
res2: Class[Me[_]] = class java.lang.Object
```

The fact that the literal type in the last example is an `ExistentialType` rather than a `TypeRef` tricks erasure into erasing it to an `ErasedValueType`.

**Comment by hrhino** [comment]
"Fixing" this and running partest reveals that this behavior is [mandated by the tests](https://github.com/scala/scala/blob/c2a5883891a68180b143eb462c8b0cebc8d3b021/test/files/run/valueclasses-classtag-existential.check#L1) somehow. Going to look into that before making a PR.

**Comment by hrhino** [comment]
This is fixed already in scala-js: scala-js/scala-js#2801

**Comment by hrhino** [comment]
scala/scala#6131
